src/System.Private.CoreLib/shared/System/Number.Parsing.cs

```diff
@@ -1910,5 +1916,89 @@ internal static float NumberToSingle(ref NumberBuffer number)
             float result = BitConverter.Int32BitsToSingle((int)(bits));
             return number.IsNegative ? -result : result;
         }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        internal static bool IsHexPrefix(ReadOnlySpan<char> str, int i) =>
```

Comment:
These 3 methods should be left in Guid.cs - Number.Parsing.cs is already quite big and these methods aren't used anywhere else.